islandora_solr_config/includes/rss_results.inc

```diff
@@ -191,7 +191,53 @@ class IslandoraSolrResultsRSS extends IslandoraSolrResults {
     $result['url'] = $base_url;
     $result['description'] = t('Aggregated search results of: @query', array('@query' => $query));
     $result['langcode'] = NULL;
+
+    $feed_url = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? "https" : "http") . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]";
```

Comment:
The output of `request_path()` is just `islandora/search`. Same result with `current_path()`. I don't know that it's possible to get this with a Drupal function.

Reply:
Passing a local path like `islandora/search` to the url function will get you a complete URL. So you  can make your urls using something like: 
```php
$page_params = drupal_get_query_parameters();
$args = array(
  'query' => array(
    'page' => $page_prev,
  ) + $page_params,
);
$link_prev = url(request_path(), $args);
```

Reply:
Hmm, your code here gets me the base feed path but puts the page query in the wrong place and without the ampersand:

`/islandora/search?page=1&type=dismax&islandora_solr_search_navigation=0&solr_profile=rss`

Reply:
Checkout this page on URL [query string](https://en.wikipedia.org/wiki/Query_string). The first parameter never has an ampersand, and the order doesn't matter. Thats why we should be using drupal functions to do this parsing rather than manual parsing, you can't rely on page always having an ampersand before it.

Reply:
Cool -- I'm getting somewhere now. Just having some trouble with the initial $feed_url replacement.

Reply:
The query stuff is working, except for the $feed_url portion - I'm getting stuck.

```
if (isset($page_params['page'])) {
  // Problem is here
      $page = array(array() + $page_params);
      $link_first = url($base_url . "/" . request_path(), $page);
    }
    else {
      $link_first = $feed_url;
    }
```

I can't figure out how to get the request_uri() with the `page` parameter removed. Do you know of a way to do this without regex?

Reply:
I don't understand what this code is trying to do.

This would get you a link to the first page of results:
```php
$page_params = drupal_get_query_parameters();
$page_number = isset($page_params['page']) ? $page_params['page'] : 1;

$link_first = url(
  request_path(), 
  array(
    'absolute' => TRUE,
    'query' => array('page' => 1) + $page_params,
  ),
);
```

As shown in that code snippet, you don't need to use `$base_url` with the `url` function.